batcher/aligned-sdk/src/sdk.rs

```diff
@@ -0,0 +1,423 @@
+use crate::errors;
+use crate::eth;
+use crate::models::{
+    AlignedVerificationData, BatchInclusionData, Chain, VerificationCommitmentBatch,
+    VerificationData, VerificationDataCommitment,
+};
+use sha3::{Digest, Keccak256};
+use std::sync::Arc;
+use tokio::{net::TcpStream, sync::Mutex};
+use tokio_tungstenite::tungstenite::Message;
+use tokio_tungstenite::{MaybeTlsStream, WebSocketStream};
+
+use log::{debug, error};
+
+use ethers::providers::{Http, Provider};
+use ethers::utils::hex;
+use futures_util::{
+    future,
+    stream::{SplitSink, SplitStream},
+    SinkExt, StreamExt, TryStreamExt,
+};
+
+/// Submits the proofs to the batcher to be verified and returns a vector of Aligned verification data.
+/// # Arguments
+/// * `ws_write` - A mutex-protected split sink to write messages to the websocket.
+/// * `ws_read` - A split stream to read messages from the websocket.
+/// * `verification_data` - A vector of verification data to be submitted to the batcher.
+/// # Returns
+/// * A vector of Aligned verification data.
+/// # Errors
+/// * If the verification data vector is empty.
+/// * If there is an error serializing the verification data.
+/// * If there is an error sending the message to the websocket.
+/// * If there is an error receiving the response from the websocket.
+/// * If there is an error closing the websocket.
+/// * If there is an error deserializing the response from the websocket.
+/// * If the connection was closed before receiving all messages.
+pub async fn submit(
+    ws_write: Arc<Mutex<SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, Message>>>,
+    ws_read: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>,
+    verification_data: Vec<VerificationData>,
+) -> Result<Option<Vec<AlignedVerificationData>>, errors::SubmitError> {
+    if verification_data.is_empty() {
+        return Err(errors::SubmitError::MissingParameter(
+            "verification_data".to_string(),
+        ));
+    }
+    let ws_write_clone = ws_write.clone();
+    let mut ws_write = ws_write.lock().await;
+
+    // The sent verification data will be stored here so that we can calculate
+    // their commitments later.
+    let mut sent_verification_data: Vec<VerificationData> = Vec::new();
+
+    for verification_data in verification_data.iter() {
+        let json_data =
+            serde_json::to_string(&verification_data).map_err(errors::SubmitError::SerdeError)?;
+        ws_write
+            .send(Message::Text(json_data.to_string()))
+            .await
+            .map_err(errors::SubmitError::ConnectionError)?;
+        sent_verification_data.push(verification_data.clone());
+        debug!("Message sent...");
+    }
+
+    drop(ws_write);
```

Comment:
This should be in a different scope, instead of manually calling drop. As an alternative you can create a helper function that takes in the mutex and the necessary params and call it.

Reply:
Solved in [0d31dda](https://github.com/yetanotherco/aligned_layer/pull/459/commits/0d31dda049b441f8ff50fd9e8b854ea4f64e3fed)